app/assets/javascripts/govuk_publishing_components/components/single-page-notification-button.js

```diff
@@ -28,7 +29,12 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
           // test that the html returned contains the button component; if yes, swap the button for the updated version
           var responseHasButton = html.querySelector('form.gem-c-single-page-notification-button .gem-c-single-page-notification-button__submit')
           if (responseHasButton) {
-            this.$module.outerHTML = newButton
+            var parent = this.$module.parentElement
+            var updatedModule = parent.insertBefore(html.firstChild, this.$module)
+            parent.removeChild(this.$module)
+            //  re-initialise click tracking on the updated element
+            var trackDetails = new window.GOVUK.Modules.GemTrackClick()
+            trackDetails.start($(updatedModule))
```

Comment:
I think I understand why the tracking is being reinitialised here but I wonder if there's an alternative approach. As long as some of the original HTML is present after the JS does the bit above, the tracking script should still work. If the entirety of the markup is rebuilt then I'd argue there might be a case for retaining the wrapping element if only to not have to reinitialise like this?

Related question - where is the tracking initialised in the first place? I can't see it - it seems odd that this code reinitialises it but doesn't start it in the first place (edit: just spotted, it's applied in the helper).

Reply:
I have applied `data-module=gem-track-click` to a wrapper as per your suggestion, which eliminates the requirement to reinitialise tracking as I had done initially.